cpp/box/Box.h

```diff
@@ -381,6 +382,53 @@ class Box
         });
     }
 
+    //! Calculate distance between two points using boundary conditions
+    /*! \param p_i Particle position
+        \param p_j Particle position to calculate distances from
+    */
+    float computeDistance(const vec3<float>& p_i, const vec3<float>& p_j) const
+    {
+            vec3<float> v_ij = wrap(p_j - p_i);
+            return std::sqrt(dot(v_ij, v_ij));
+    }
+
+    //! Calculate distance between a set of points and query points using boundary conditions
+    /*! \param points Particle positions
+        \param query_points Particle position to calculate distances from
+        \param dist Distances between points and query_points
+        \param Nvecs The number of points to calculate distances between
+    */
+    void computeDistances(vec3<float>* points, vec3<float>* query_points, float *dist, unsigned int Nvecs) const
+    {
+        util::forLoopWrapper(0, Nvecs, [=](size_t begin, size_t end) {
+            for (size_t i = begin; i < end; ++i)
+            {
+                dist[i] = computeDistance(points[i], query_points[i]);
+            }
+        });
+    }
+
+    //! Calculate distance between a set of points and query points using boundary conditions
+    /*! \param points Particle positions
+        \param query_points Particle position to calculate distances from
+    */
+    void computeAllDistances(vec3<float>* points, vec3<float>* query_points,
+        float* dist, unsigned int Nvecs, unsigned int Mvecs) const
```

Comment:
I would like `Nvecs` to be named so that it's clear it corresponds to either points or query points. e.g. `N_points` or `N_query_points`. Similarly, `Mvecs` should be the other one.